core-bundle/contao/widgets/KeyValueWizard.php

```diff
@@ -73,7 +73,7 @@ public function validate()
 				$options[$key]['key'] = trim($option['key']);
 				$options[$key]['value'] = trim($option['value']);
 
-				if ($options[$key]['key'])
+				if ($options[$key]['key'] || ($this->allowEmptyKeys && $options[$key]['value']))
```

Comment:
```suggestion
				if ($options[$key]['key'] || ($this->allowEmptyKeys && $options[$key]['value'] !== ''))
```
Otherwise the string `'0'` would be seen as empty.